--- a/net/minecraft/src/BlockDoor.java
+++ b/net/minecraft/src/BlockDoor.java
@@ -129,8 +129,8 @@
 		return this.getFullMetadata(var1, var2, var3, var4) & 3;
 	}
 
-	public boolean isDoorOpen(IBlockAccess var1, int var2, int var3, int var4) {
-		return (this.getFullMetadata(var1, var2, var3, var4) & 4) != 0;
+	public boolean isDoorOpen(IBlockAccess blockAcess, int x, int y, int z) {
+		return (this.getFullMetadata(blockAcess, x, y, z) & 4) != 0;
 	}
 
 	public void setDoorRotation(int var1) {
@@ -191,51 +191,60 @@
 		this.blockActivated(var1, var2, var3, var4, var5);
 	}
 
-	public boolean blockActivated(World var1, int var2, int var3, int var4, EntityPlayer var5) {
+	public boolean blockActivated(World world, int x, int y, int z, EntityPlayer player) {
 		if(this.material == Material.iron) {
 			return true;
 		} else {
-			int var6 = this.getFullMetadata(var1, var2, var3, var4);
+			int var6 = this.getFullMetadata(world, x, y, z);
 			int i7 = var6 & 7;
 			i7 ^= 4;
 			if((var6 & 8) != 0) {
-				var1.setBlockMetadataWithNotify(var2, var3 - 1, var4, i7);
-				var1.markBlocksDirty(var2, var3 - 1, var4, var2, var3, var4);
+				world.setBlockMetadataWithNotify(x, y - 1, z, i7);
+				world.markBlocksDirty(x, y - 1, z, x, y, z);
 			} else {
-				var1.setBlockMetadataWithNotify(var2, var3, var4, i7);
-				var1.markBlocksDirty(var2, var3, var4, var2, var3, var4);
+				world.setBlockMetadataWithNotify(x, y, z, i7);
+				world.markBlocksDirty(x, y, z, x, y, z);
 			}
 
-			if(Math.random() < 0.5D) {
-				var1.playSoundEffect((double)var2 + 0.5D, (double)var3 + 0.5D, (double)var4 + 0.5D, "random.door_open", 1.0F, var1.rand.nextFloat() * 0.1F + 0.9F);
-			} else {
-				var1.playSoundEffect((double)var2 + 0.5D, (double)var3 + 0.5D, (double)var4 + 0.5D, "random.door_close", 1.0F, var1.rand.nextFloat() * 0.1F + 0.9F);
+			ModSettings modSettings = ModSettings.getInstance();
+			
+			if (modSettings.getOptionValue(ModSetting.DOOR_SOUND) == ModSettingOptions.DOOR_SOUND_OPEN_CLOSE) {
+				if(this.isDoorOpen(world, x, y, z)) {
+					world.playSoundEffect((double)x + 0.5D, (double)y + 0.5D, (double)z + 0.5D, "random.door_close", 1.0F, world.rand.nextFloat() * 0.1F + 0.9F);				
+				} else {
+					world.playSoundEffect((double)x + 0.5D, (double)y + 0.5D, (double)z + 0.5D, "random.door_open", 1.0F, world.rand.nextFloat() * 0.1F + 0.9F);
+				}
+			} else if (modSettings.getOptionValue(ModSetting.DOOR_SOUND) == ModSettingOptions.DOOR_SOUND_RANDOM) {
+				if(Math.random() < 0.5D) {
+					world.playSoundEffect((double)x + 0.5D, (double)y + 0.5D, (double)z + 0.5D, "random.door_close", 1.0F, world.rand.nextFloat() * 0.1F + 0.9F);				
+				} else {
+					world.playSoundEffect((double)x + 0.5D, (double)y + 0.5D, (double)z + 0.5D, "random.door_open", 1.0F, world.rand.nextFloat() * 0.1F + 0.9F);
+				}
 			}
-
+			
 			return true;
 		}
 	}
 
-	public void onPoweredBlockChange(World var1, int var2, int var3, int var4, boolean var5) {
-		int var6 = this.getFullMetadata(var1, var2, var3, var4);
+	public void onPoweredBlockChange(World world, int var2, int var3, int var4, boolean var5) {
+		int var6 = this.getFullMetadata(world, var2, var3, var4);
 		boolean var7 = (var6 & 4) != 0;
 		if(var7 != var5) {
 			int i8 = var6 & 7;
 			i8 ^= 4;
 			if((var6 & 8) != 0) {
-				var1.setBlockMetadataWithNotify(var2, var3 - 1, var4, i8);
-				var1.markBlocksDirty(var2, var3 - 1, var4, var2, var3, var4);
+				world.setBlockMetadataWithNotify(var2, var3 - 1, var4, i8);
+				world.markBlocksDirty(var2, var3 - 1, var4, var2, var3, var4);
 			} else {
-				var1.setBlockMetadataWithNotify(var2, var3, var4, i8);
-				var1.markBlocksDirty(var2, var3, var4, var2, var3, var4);
+				world.setBlockMetadataWithNotify(var2, var3, var4, i8);
+				world.markBlocksDirty(var2, var3, var4, var2, var3, var4);
 			}
 
 			if(Math.random() < 0.5D) {
-				var1.playSoundEffect((double)var2 + 0.5D, (double)var3 + 0.5D, (double)var4 + 0.5D, "random.door_open", 1.0F, var1.rand.nextFloat() * 0.1F + 0.9F);
+				world.playSoundEffect((double)var2 + 0.5D, (double)var3 + 0.5D, (double)var4 + 0.5D, "random.door_open", 1.0F, world.rand.nextFloat() * 0.1F + 0.9F);
 			} else {
-				var1.playSoundEffect((double)var2 + 0.5D, (double)var3 + 0.5D, (double)var4 + 0.5D, "random.door_close", 1.0F, var1.rand.nextFloat() * 0.1F + 0.9F);
+				world.playSoundEffect((double)var2 + 0.5D, (double)var3 + 0.5D, (double)var4 + 0.5D, "random.door_close", 1.0F, world.rand.nextFloat() * 0.1F + 0.9F);
 			}
-
 		}
 	}
 
