--- a/net/minecraft/src/EntityRenderer.java
+++ b/net/minecraft/src/EntityRenderer.java
@@ -554,8 +554,8 @@
 			ScaledResolution var7 = new ScaledResolution(this.mc.options, this.mc.displayWidth, this.mc.displayHeight);
 			int var8 = var7.getScaledWidth();
 			int var9 = var7.getScaledHeight();
-			int var10 = Mouse.getX() * var8 / this.mc.displayWidth;
-			int var11 = var9 - Mouse.getY() * var9 / this.mc.displayHeight - 1;
+			int var10 = this.mc.scaleMouseX(Mouse.getX()) * var8 / this.mc.displayWidth;
+			int var11 = var9 - this.mc.scaleMouseY(Mouse.getY()) * var9 / this.mc.displayHeight - 1;
 			if(this.mc.theWorld != null) {
 				this.renderWorld(var1);
 				if(this.mc.options.hideGUI != 1 || this.mc.currentScreen != null) {
