--- a/net/minecraft/src/ItemPickaxe.java
+++ b/net/minecraft/src/ItemPickaxe.java
@@ -3,17 +3,17 @@
 public class ItemPickaxe extends ItemTool {
 	private static Block[] blocksEffectiveAgainst = new Block[]{Block.cobblestone, Block.stairDouble, Block.stairSingle, Block.stone, Block.cobblestoneMossy, Block.oreIron, Block.blockSteel, Block.oreCoal, Block.blockGold, Block.oreGold, Block.oreDiamond, Block.blockDiamond, Block.ice, Block.oreRedstone, Block.oreRedstoneGlowing, Block.minecartTrack};
 
-	public ItemPickaxe(int var1, String var2) {
-		super(var1, 2, var2, blocksEffectiveAgainst);
+	public ItemPickaxe(int id, String material) {
+		super(id, 2, material, blocksEffectiveAgainst);
 	}
 
-	public boolean canHarvestBlock(Block var1) {
-		return var1 == Block.obsidian ? this.harvestLevel == 3 : (var1 != Block.blockDiamond && var1 != Block.oreDiamond ? (var1 != Block.blockGold && var1 != Block.oreGold ? (var1 != Block.blockSteel && var1 != Block.oreIron ? (var1 != Block.oreRedstone && var1 != Block.oreRedstoneGlowing ? (var1.material == Material.rock ? true : var1.material == Material.iron) : this.harvestLevel >= 2) : this.harvestLevel >= 1) : this.harvestLevel >= 2) : this.harvestLevel >= 2);
+	public boolean canHarvestBlock(Block block) {
+		return block == Block.obsidian ? this.harvestLevel == 3 : (block != Block.blockDiamond && block != Block.oreDiamond ? (block != Block.blockGold && block != Block.oreGold ? (block != Block.blockSteel && block != Block.oreIron ? (block != Block.oreRedstone && block != Block.oreRedstoneGlowing ? (block.material == Material.rock ? true : block.material == Material.iron) : this.harvestLevel >= 2) : this.harvestLevel >= 1) : this.harvestLevel >= 2) : this.harvestLevel >= 2);
 	}
 
-	public float getStrVsBlock(ItemStack itemStack, Block var1) {
-		if(var1 == null || var1.material != Material.iron && var1.material != Material.rock) {
-			return super.getStrVsBlock(itemStack, var1);
+	public float getStrVsBlock(ItemStack itemStack, Block block) {
+		if(block == null || block.material != Material.iron && block.material != Material.rock) {
+			return super.getStrVsBlock(itemStack, block);
 		}
 
 		return this.efficiencyOnProperMaterial;
